scripts/tf_cnn_benchmarks/benchmark_cnn.py

```diff
@@ -3253,6 +3267,10 @@ def forward_pass_and_gradients():
         # All-reduce gradients using Horovod.
         grads = [hvd.allreduce(grad, average=False, device_dense=horovod_device)
                  for grad in grads]
+      elif self.params.variable_update == 'kungfu':
+        from kungfu.optimizers.core import lazy_load_op_lib
+        _op_lib = lazy_load_op_lib()
+        grads = [_op_lib.all_reduce(grad) for grad in grads]
```

Comment:
It would be helpful to increase the level of abstraction here to be able to use optimizers such as Ako.